### ISSUE TITLE
[x64] Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#18963]
Failure in CI in x64 corefx minopts tests. E.g., https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64/job/jitstress/job/x64_checked_windows_nt_corefx_minopts/581/consoleText

There are 60 occurrences of this assert in this run.

```
----- start  9:56:58.02 ===============  To repro directly: ===================================================== 
  pushd D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\tests\System.Buffers.Tests\netcoreapp-Windows_NT-Release-x64\
  set COMPlus_TieredCompilation=0 
  call D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\testhost\netcoreapp-Windows_NT-Release-x64\\dotnet.exe xunit.console.netcore.exe System.Buffers.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
  popd
  ===========================================================================================================
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: System.Buffers.Tests
  [Microsoft (R) .NET Core Xml Serialization Generation Utility, Version 2.0.0-preview1]
  Discovered:  System.Buffers.Tests
  Copyright (C) Microsoft Corporation. All rights reserved.
  Starting:    System.Buffers.Tests
  Serialization Code File Name: D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\tests\Microsoft.XmlSerializer.Generator.Tests\netcoreapp-Windows_NT-Release-x64\Microsoft.XmlSerializer.Generator.Tests.XmlSerializers.cs.
  Discovered:  System.Collections.Concurrent.Tests
  Generated serialization code for assembly D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\tests\Microsoft.XmlSerializer.Generator.Tests\netcoreapp-Windows_NT-Release-x64\Microsoft.XmlSerializer.Generator.Tests.dll --> 'D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\tests\Microsoft.XmlSerializer.Generator.Tests\netcoreapp-Windows_NT-Release-x64\Microsoft.XmlSerializer.Generator.Tests.XmlSerializers.cs'.
  
  Assert failure(PID 3144 [0x00000c48], Thread: 4960 [0x1360]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Uri:PrivateParseMinimal():int:this' (IL size 1208)
  
      File: d:\j\workspace\x64_checked_w---f0b6fff8\src\jit\flowgraph.cpp Line: 21774
      Image: D:\j\workspace\x64_checked_w---f0b6fff8\_\fx\bin\testhost\netcoreapp-Windows_NT-Release-x64\dotnet.exe
  
  ----- end  9:57:05.41 ----- exit code 123456789 ----------------------------------------------------------

```

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
Also shows up in PMI of System.Private.Uri.dll:

```
D:\repos\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe d:\repos\jitutils\bin\pmi.dll prepall-quiet D:\repos\coreclr\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\System.Private.Uri.dll

Assert failure(PID 16088 [0x00003ed8], Thread: 12664 [0x3178]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Uri:PrivateParseMinimal():int:this' (IL size 1208)

    File: d:\repos\coreclr\src\jit\flowgraph.cpp Line: 21774
    Image: D:\repos\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

**Comment by AndyAyersMS** [comment]
Seems to be in the post-phase of the ObjectAllocator. Assert is that the tree node 0002 belongs to more than one tree:
```
               [000009] ------------              *  STMT      void  (IL 0x000...0x00E)
               [001018] ---XG+------              |     /--*  CAST      int <- int
(#1)           [000002] ---XG+------              |     |  \--*  IND       ushort
               [001016] -----+------              |     |     |  /--*  CNS_INT   long   48 field offset Fseq[_flags]
               [001017] -----+------              |     |     \--*  ADD       byref 
               [000001] -----+------              |     |        \--*  LCL_VAR   ref    V00 this         
               [000005] ---XG+------              |  /--*  AND       int   
(#2)           [000002] ---XG+------              |  |  \--*  IND       ushort
               [001016] -----+------              |  |     |  /--*  CNS_INT   long   48 field offset Fseq[_flags]
               [001017] -----+------              |  |     \--*  ADD       byref 
               [000001] -----+------              |  |        \--*  LCL_VAR   ref    V00 this         
               [000008] -A-XG+------              \--*  ASG       int   
               [000007] D----+-N----                 \--*  LCL_VAR   int    V01 loc0         
```

Seems likely to be related to #18916.

@erozen PTAL

**Comment by erozenfeld** [comment]
This was caused by my change https://github.com/dotnet/coreclr/commit/c0bad3c66218f61dd3e4f151b9d2eae22c6d88d2
and already caught in corefx.
There is a PR to revert that change #18975 and I'm working on a fix.

**Comment by erozenfeld** [comment]
#18975 was just merged.

**Comment by erozenfeld** [comment]
Fixed by #18995.
